Source/Particles/ERFPCEvolve.cpp

```diff
@@ -98,8 +98,7 @@ void ERFPC::AdvectWithFlow ( MultiFab*                           a_umac,
             bool use_terrain = (a_z_height != nullptr);
             auto zheight = use_terrain ? (*a_z_height)[grid].array() : Array4<Real>{};
```

Comment:
@AMLattanzi how does this make `zheight` available on the GPU?

Reply:
This is due to C++ lambda capture. If a C++ lambda uses `[=]`, it will make a copy of all the variables used in the lambda's function body. These variables are part of the lambda function object. In the case of GPU, the CUDA runtime will copy the lambda function to GPU at kernel launch.

Reply:
Thanks, @WeiqunZhang! Makes sense now.